gateleen-hook/src/main/java/org/swisspush/gateleen/hook/HookHandler.java

```diff
@@ -1032,23 +1041,33 @@ private void handleListenerRegistration(final HttpServerRequest request) {
             final String listenerStorageUri = hookRootUri + HOOK_LISTENER_STORAGE_PATH + getUniqueListenerId(request.uri());
 
             // Extract expireAfter from the registration header.
-            final String myExpireAfter = request.headers().get( X_EXPIRE_AFTER );
             final String expirationTime;
-            if( "-1".equals(myExpireAfter) ){
-                expirationTime = EXPIRATION_TIME_NEVER;
-            }else{
-                Integer expireAfter = ExpiryCheckHandler.getExpireAfter(request.headers());
-                if (expireAfter == null) {
+
+            { // Evaluate expiration time
```

Comment:
I would suggest to extract this block of code in a separate method rather than using an anonymous code block to restrict variable scope.